book.rb

```diff
@@ -0,0 +1,14 @@
+class Book
+  attr_accessor :title, :author
+  attr_reader :rentals
+
+  def initialize(title, author)
+    @title = title
+    @author = author
+    @rentals = []
+  end
+
+  def add_rental(rental)
+    @rentals << rental
+  end
```

Comment:
- [ ] I can see that this method is used in the Rental class as way of pushing the rental object to the rentals variable in this class. But this does not completely set up the `many-to-many` association between book and person using the rental as intermediate. We still need a new method which when called on a `book` object, with a `date` and `person` parameters, will create a new record of rental.

  ## HINT
  - You'll need to import the Rental class here as well.
  ```
  def create_rental(person, date)
    Rental.new(date, person, self)
  end
  ```

Reply:
According to the requirements (Modify the constructor of Rental so Book and Person are set in it.). I think that means the book, date and person are created in the constructor of rental. Now to create a many-to-many association we created a function in `Book `and `Person` which will add the new rental to each of them. That's how we learned it in the previous lesson. If you try my code, you will see everything is working perfectly. 
![image](https://user-images.githubusercontent.com/81693905/174062778-e86d1e5d-9ef2-4311-a8fc-14c7eadc3518.png)

Reply:
If there is a `has-many-through` relationship between Book and Person through Rental, it means that given both objects, one should be able to create a rental. You have created an intermediate table here Rental, but Book and Person need their own respective methods that allows one create a Rental.

The middle class has been created, the other classes need to use it on the own instances.